### ISSUE TITLE
isSigned returning true for unsigned transactions

### ISSUE THREAD

**Issue by alexfmpe** [ethereumjs/ethereumjs-vm#1041]
Creating a transaction with `fromTxData` **without** the signature fields leads to them being initialized 
```
v: <BN: 0>,
r: <BN: 0>,
s: <BN: 0>,
```

which makes `isSigned` return `true`
https://github.com/ethereumjs/ethereumjs-vm/blob/3588bd8dbc744eec1b58c4014cef3838d0a539aa/packages/tx/src/transaction.ts#L359
 
```
> !!0
false
> !!(new BN(0))
true
```

**Comment by ryanio** [comment]
ah good catch, they should indeed not be initialized if missing, so these lines in the constructor:

```typescript
    this.v = new BN(toBuffer(v))
    this.r = new BN(toBuffer(r))
    this.s = new BN(toBuffer(s))
```

should change to:

```typescript
    this.v = v ? new BN(toBuffer(v)) : undefined
    this.r = r ? new BN(toBuffer(r)) : undefined
    this.s = s ? new BN(toBuffer(s)) : undefined
```

**Comment by holgerd77** [comment]
Oh, that's an important one, I'll add a respective label here and we should release relatively soon once we have merged the respective PR from @ryanio (thanks 😄 ). Think we are generally at a point where it makes sense to target a new round of releases after the initial v5 releases.

**Comment by holgerd77** [comment]
Closed by #1042
